genie_registry/clinical.py

```diff
@@ -77,6 +77,92 @@ def _check_year(clinicaldf: pd.DataFrame, year_col: int, filename: str,
     return error
 
 
+def _check_int_dead_consistency(clinicaldf: pd.DataFrame) ->  str:
+    """Check if vital status interval and dead column are consistent
+
+    Args:
+        clinicaldf: Clinical Data Frame
+
+    Returns:
+        Error message if values and inconsistent or blank string
+    """
+    cols = ["INT_DOD", "DEAD"]
+    for col in cols:
+        # Return empty string is columns don't exist because this error
+        # is already handled.
+        if not process_functions.checkColExist(clinicaldf, col):
+            return ''
+    is_dead = clinicaldf['DEAD'].astype(str) == "True"
+    is_alive = clinicaldf['DEAD'].astype(str) == "False"
+    allowed_str = ['Unknown', 'Not Collected', 'Not Applicable',
+                   "Not Released"]
+    is_str = clinicaldf['DEAD'].isin(allowed_str)
+    # Check that all string values are equal each other
+    is_equal = all(
+        clinicaldf.loc[is_str, "DEAD"] == clinicaldf.loc[is_str, "INT_DOD"]
+    )
+    # If dead, int column can't be Not Applicable
+    # If alive, int column can't have values
+    if (any(clinicaldf.loc[is_dead, 'INT_DOD'] == "Not Applicable") or
+            not all(clinicaldf.loc[is_alive, 'INT_DOD'].isin(allowed_str)) or
+            not is_equal):
+        return (
+            "Patient: you have inconsistent values in "
+            "INT_DOD, YEAR_DEATH, DEAD\n"
```

Comment:
Would suggest changing the comment to explicitly indicate the DEAD variable needs to be changed.  And would you be able to extract the indexes of the rows and return the relevant patients that violate the rules?  I think it may be difficult for users to identify the source rows of the inconsistency without the patient ids.  Maybe something like:

"Patient: DEAD variable value is inconsistent with INT_DOD for the following patients: GENIE-xxx-xxxx, GENIE-xxx-xxxx"